### PR TITLE
Proposal for tweaks to radio widget flow types

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
This PR introduces a 'RadioChoiceWithMetadata' type to model the shape of props.choices in radio/widget.jsx without having to modify PerseusRadioChoice in perseus-types.js.

This PR is only for illustration purposes and shouldn't be merged.

Issue: None

## Test plan:
- yarn flow

### CODE REVIEW

File: packages/perseus/src/types.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import * as React from "react";
 import type {SerializedHighlightSet} from "./components/highlighting/types.js";
 import type {ILogger} from "./logging/log.js";
 import type {Item} from "./multi-items/item-types.js";
-import type {PerseusWidget} from "./perseus-types.js";
+import type {PerseusWidget, PerseusRadioChoice} from "./perseus-types.js";
```
Comment (handeyeco) [inline packages/perseus/src/types.js:12]: What's the difference between `types.js` and `perseus-types.js`? I imagine one is externally facing and one is internally facing?

Comment (jeremywiebe) [reply packages/perseus/src/types.js:12]: `perseus-types.js` is the file that contains mostly types that model the JSON that gets published (ie. Perseus data). `types.js` is the file containing types that Perseus uses (internally, for the most part).

Comment (kevinbarabash) [reply packages/perseus/src/types.js:12]: We should probably add a comment at the top of `perseus-types.js` and/or rename the file to `perseus-json-types.js` or something like that to make it more obvious that these types describe the data that's in data store.

**Comment by kevinbarabash** [comment]
These changes have been rolled into #313.

**Pull request closed without merging**